src/main/java/com/jfrog/ide/idea/configuration/GlobalSettings.java

```diff
@@ -52,11 +52,23 @@ public XrayServerConfigImpl getXrayConfig() {
         return this.xrayConfig;
     }
 
+    // Required for deserialization.
     public void setXrayConfig(XrayServerConfigImpl xrayConfig) {
+        this.xrayConfig.setUrl(xrayConfig.getUrl());
+        this.xrayConfig.setUsername(xrayConfig.getUsername());
+        this.xrayConfig.setCredentials(xrayConfig.getCredentialsFromPasswordSafe());
+        this.xrayConfig.setExcludedPaths(xrayConfig.getExcludedPaths());
+    }
+
+    public void updateConfig(XrayServerConfigImpl xrayConfig) {
+        if (this.xrayConfig.getUrl() != null && !this.xrayConfig.getUrl().equals(xrayConfig.getUrl())) {
+            this.xrayConfig.removeCredentialsFromPasswordSafe();
```

Comment:
At first glance, it looks like updateConfig removes the credentials from the passwordSafe and set them again. Are you sure the deletion is necessary?

Reply:
IMO it is necessary.
If a certain URL was previously saved to the password safe, and then the URL changes -> I think we should remove the old one.